src/main/java/org/jenkinsci/plugins/gitclient/GitCredentialBindings.java

```diff
@@ -29,43 +29,47 @@
-                        //Comparing and changing git home for git tool
-                        gitTool = gitTool.forNode(currentNode, listener);
-                        return gitTool.getGitExe();
+            //Check node is not null
+            if (currentNode != null) {
+                if (currentNode.getNodeName().equals("")) {
+                    //It's on the master node
+                    toolType = Jenkins.get().getDescriptorByType(GitTool.DescriptorImpl.class);
+                    if (toolType != null) {
+                        ToolInstallation[] toolInstalled = toolType.getInstallations();
+                        for (ToolInstallation t : toolInstalled) {
+                            if (t.getClass().equals(GitTool.class)) {
+                                return ((GitTool) t).getGitExe();
+                            }
+                        }
                     }
-                }
-            }
-            if (toolType != null) {
-                ToolInstallation[] toolInstalled = toolType.getInstallations();
-                for (ToolInstallation t : toolInstalled) {
-                    if (t.getClass().equals(GitTool.class)) {
-                        return ((GitTool) t).getGitExe();
+                } else {
+                    //It's on an agent type
+                    ToolLocationNodeProperty nodeToolLocation = currentNode.getNodeProperty(ToolLocationNodeProperty.class);
+                    //Tool location property is configured
+                    if (nodeToolLocation != null) {
+                        //Get Tool locations(could be many)
+                        List<ToolLocationNodeProperty.ToolLocation> toolList = nodeToolLocation.getLocations();
+                        //Iteration
+                        for (ToolLocationNodeProperty.ToolLocation tool : toolList) {
+                            //If 'Default' Git Tool is chosen on an agent then all the Git Tools installations added
+                            // to the master nodes will be available on the agent as well
+                            //Check not an instance of jgit or jgitapache using tool descriptor not type key
+                            if (tool.getType().clazz.equals(GitTool.class)) {
+                                //Check for properties
+                                GitTool gitTool = new GitTool(tool.getName(), tool.getHome(), null);
+                                //node specific git home for git tool
+                                gitTool = gitTool.forNode(currentNode, listener);
+                                return gitTool.getGitExe();
+                            }
+                        }
                     }
                 }
+                listener.getLogger().println("No Git Tool is configured on the current node");
             }
-        } else {
-            return null;
+            listener.getLogger().println("Is this node available...");
```

Comment:
I'm guessing that this is just debugging information and can be removed before we merge?